Source/Core/Core/AchievementManager.cpp

```diff
@@ -1229,16 +1304,33 @@ u32 AchievementManager::MemoryPeeker(u32 address, u8* buffer, u32 num_bytes, rc_
 {
   if (buffer == nullptr)
     return 0u;
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+  auto& instance = AchievementManager::GetInstance();
+  if (instance.m_dll_found)
+  {
+    std::lock_guard lg{instance.m_memory_lock};
+    if (address + num_bytes >= instance.m_cloned_memory.size())
```

Comment:
This risks overflowing.

```suggestion
    if (u64(address) + num_bytes >= instance.m_cloned_memory.size())
```